src/engine/openspaceengine.cpp

```diff
@@ -1546,6 +1584,144 @@ void OpenSpaceEngine::toggleShutdownMode() {
+                global::scriptEngine->queueScript(
+                    geoScript,
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        },
+        p.camera.value()
+    );
+}
+
+void OpenSpaceEngine::setFromProfile_deltaTimes(const Profile& p) {
+    global::timeManager->setDeltaTimeSteps(p.deltaTimes);
+}
+
+void OpenSpaceEngine::setFromProfile_modules(const Profile& p)
+{
+    for (Profile::Module mod : p.modules) {
+        const std::vector<OpenSpaceModule*>& m = global::moduleEngine->modules();
+        const auto it = std::find_if(m.begin(), m.end(),
+            [&mod](const OpenSpaceModule* moduleSearch) {
+                return (moduleSearch->identifier().compare(mod.name) == 0);
+            });
+        if (it != m.end()) {
+            if (mod.loadedInstruction.has_value()) {
+                global::scriptEngine->queueScript(
+                    mod.loadedInstruction.value(),
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        }
+        else {
+            if (mod.notLoadedInstruction.has_value()) {
+                global::scriptEngine->queueScript(
+                    mod.notLoadedInstruction.value(),
+                    scripting::ScriptEngine::RemoteScripting::Yes
+                );
+            }
+        }
+    }
+}
+
+void OpenSpaceEngine::setFromProfile_actions(const Profile& p)
+{
+    for (Profile::Action a : p.actions) {
+        if (a.identifier.empty()) {
+            LERROR("Identifier must to provided to register action");
+        }
+        if (global::actionManager->hasAction(a.identifier)) {
+            LERROR(fmt::format("Action for identifier '{}' already existed & registered",
+                a.identifier));
+        }
+        if (a.script.empty()) {
+            LERROR(fmt::format("Identifier '{}' doesn't provide a Lua command to execute",
+                a.identifier));
+        }
+        interaction::Action action;
+        action.identifier = a.identifier;
+        action.command = a.script;
+        if (!a.name.empty()) {
+            action.name = a.name;
+        }
+        if (!a.documentation.empty()) {
+            action.documentation = a.documentation;
+        }
+        if (!a.guiPath.empty()) {
+            action.guiPath = a.guiPath;
+        }
+        action.synchronization = interaction::Action::IsSynchronized(a.isLocal);
+        global::actionManager->registerAction(std::move(action));
+    }
+}
+
+void OpenSpaceEngine::setFromProfile_keybindings(const Profile& p)
+{
+    for (Profile::Keybinding k : p.keybindings) {
+        if (k.action.empty()) {
+            LERROR("Action must not be empty");
+        }
+        if (!global::actionManager->hasAction(k.action)) {
+            LERROR(fmt::format("Action '{}' does not exist", k.action));
+        }
+        if (k.key.key == openspace::Key::Unknown) {
+            LERROR(fmt::format("Could not find key '{}'",
```

Comment:
Same as above:
```
LERROR(fmt::format(
    "Could not find key '{}'",
    std::to_string(static_cast<uint16_t>(k.key.key))
));
```

I don't think the `std::to_string` is necessary either since fmt::format can accept uint16_t and do the string conversion itself

Reply:
The std::to_string was necessary to make the compiler happy